proc-macros/src/render_server.rs

```diff
@@ -77,6 +85,15 @@ impl RpcDescription {
 			let mut sub_sig = sub.signature.clone();
 
 			sub_sig.sig.inputs.insert(1, subscription_sink);
+
+			if sub.with_context {
```

Comment:
In the proc macros I'm not convinced that should expose ConnectionContext.

Just `connection_id: usize` should be sufficient and because everything internally has connection_id can't we just require everything to expose the connection ID?